src/osd/PG.cc

```diff
@@ -3855,9 +3882,25 @@ void PG::replica_scrub(
   if (!end.is_max())
     end.pool = info.pgid.pool();
 
-  build_scrub_map_chunk(
-    map, start, end, msg->deep, msg->seed,
-    handle);
+  if (scrubber.map_state == PG::Scrubber::REG_SCRUBBING) {
+    assert(scrubber.scrubmap.objects.empty());
+    assert(scrubber.ls.empty());
+    build_scrub_map_chunk(scrubber.scrubmap, start, end, msg->seed, handle, scrubber.ls);
+  } else {
+    assert(scrubber.map_state == PG::Scrubber::DEEP_SCRUBBING);
+    get_pgbackend()->be_deep_scan_list(scrubber.scrubmap, scrubber.ls, msg->seed, handle);
```

Comment:
@athanatos It is a little confusing, but a deep scrub goes through the REG_SCRUBBING phase which gets the metadata AND the list of objects in scrubber.ls.  That list is then used after a requeue in DEEP_SCRUBBING phase to call be_deep_scan_list().